sensors/sqs_sensor.py

```diff
@@ -159,16 +158,24 @@ def _is_same_credentials(session, account_id):
         }
         required_accounts = {self._get_info(queue)[0] for queue in self.input_queues}
 
+        if not self.default_session or \
+                not _is_same_credentials(self.default_session, self.account_id,
+                                         self.default_credentials):
+            self._setup_session()
+
         for account_id in required_accounts:
             if account_id != self.account_id and account_id not in self.cross_roles:
                 continue
 
             session = self.sessions.get(account_id)
-            if not _is_same_credentials(session, account_id):
-                if account_id == self.account_id:
+            if account_id == self.account_id and self.account_id not in self.cross_roles:
```

Comment:
Some comments on these ifs/elfils would be nice. Also if you put back that check for session being `None` you can get rid of checking it here